### ISSUE TITLE
Missing functionality for quotient semigroups

### ISSUE THREAD

**Issue by ChristopherRussell** [semigroups/Semigroups#454]
I noticed a few methods not working with quotient semigroups. In particular `GeneratorsSmallest` will return output saying "this shouldn't happen!"

```
gap> S := InverseSemigroup( [ PartialPerm( [ 1, 2, 3, 4 ], [ 1, 2, 3, 4 ] ),
  PartialPerm( [ 1, 2, 3, 4, 5, 6, 7, 8, 11, 12, 13, 14, 19, 20, 29, 30 ],
     [ 2, 1, 4, 3, 6, 5, 8, 7, 12, 11, 14, 13, 20, 19, 30, 29 ] ),
  PartialPerm( [ 1, 2, 3, 4, 5, 6, 7, 8, 11, 12, 13, 14, 19, 20, 29, 30 ],
     [ 4, 3, 2, 1, 8, 7, 6, 5, 10, 9, 16, 15, 18, 17, 32, 31 ] ),
  PartialPerm( [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 16, 17, 18, 31, 32 ],
     [ 2, 1, 4, 3, 6, 5, 8, 7, 10, 9, 16, 15, 18, 17, 32, 31 ] ),
  PartialPerm( [ 1, 2, 3, 4, 21, 22, 23, 24, 25, 26, 27, 28, 33, 34, 35, 36 ],
     [ 4, 3, 2, 1, 24, 23, 22, 21, 28, 27, 26, 25, 36, 35, 34, 33 ] ),
  PartialPerm( [ 1, 2, 3, 4, 21, 22, 23, 24, 25, 26, 27, 28, 33, 34, 35, 36 ],
     [ 3, 4, 1, 2, 23, 24, 21, 22, 27, 28, 25, 26, 35, 36, 33, 34 ] ),
  PartialPerm( [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 15, 16, 17, 18, 31, 32 ],
     [ 4, 3, 2, 1, 8, 7, 6, 5, 12, 11, 14, 13, 20, 19, 30, 29 ] ),
  PartialPerm( [ 1, 2, 3, 4, 19, 20, 29, 30 ], [ 4, 3, 2, 1, 18, 17, 32, 31 ] ),
  PartialPerm( [ 1, 2, 3, 4, 22, 24, 33, 35 ], [ 4, 3, 2, 1, 23, 21, 36, 34 ] ) ] );
<inverse partial perm semigroup of rank 36 with 9 generators>
gap> cong := MinimumGroupCongruence(S);
<semigroup congruence over <inverse partial perm semigroup of rank 36 with 9 generators>
  with congruence pair (9,1)>
gap> G := S/cong;
<quotient of <semigroup congruence over <inverse partial perm semigroup of rank 36 with 9
 generators> with congruence pair (9,1)>>
gap> GeneratorsSmallest(G);
Error, EN_SEMI_ELEMENT_NUMBER_SORTED: this shouldn't happen! in
  return iter!.NextIterator( iter ); at /Users/crussell/gap/lib/coll.gi:993 called from
NextIterator( iter ) at /Users/crussell/gap/pkg/semigroups/gap/attributes/attr.gi:116 called from
<function "_GeneratorsSmallest">( <arguments> )
 called from read-eval loop at *stdin*:190

gap> Size(S/cong);
4
gap> Size(GeneratorsOfSemigroup(S/cong));
11
gap> IrredundantGeneratingSubset(S/cong);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `IsGeneratorsOfActingSemigroup' on 1 arguments at /Users/crussell/gap/lib/methsel2.g:250 called from
IsGeneratorsOfActingSemigroup( coll
 ) at /Users/crussell/gap/pkg/semigroups/gap/attributes/attr.gi:234 called from
Error( no_method_found ); at /Users/crussell/gap/lib/methsel2.g:250 called from
IsGeneratorsOfActingSemigroup( coll
 ) at /Users/crussell/gap/pkg/semigroups/gap/attributes/attr.gi:234 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *errin*:4
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
brk> quit;

gap> SmallSemigroupGeneratingSet(G);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `SmallSemigroupGeneratingSet' on 1 arguments at /Users/crussell/gap/lib/methsel2.g:250 called from
SmallSemigroupGeneratingSet( GeneratorsOfSemigroup( S )
 ) at /Users/crussell/gap/pkg/semigroups/gap/attributes/attr.gi:308 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:194
you can 'quit;' to quit to outer loop, or
you can 'return;' to continue
```

**Comment by james-d-mitchell** [comment]
Thanks @ChristopherRussell, this is also related to #370.

**Comment by james-d-mitchell** [comment]
Resolved in v4.0.3
